### PR TITLE
Assigment for week 8

### PR DESCRIPTION
**Pull request by morarodgers**
Preferred assignment for week 8

### CODE REVIEW

File: 05-JWT-Basics/preferred/middleware/authenticate.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const authHeader = req.header('Authorization')
+
+  if (!authHeader || !authHeader.startsWith('Bearer ')) {
+      res.status(401).json({message: 'Unauthorized - no token provided'})
+    }
+    const token = authHeader.split(' ')[1]
+    //console.log(token);
+
+    try {
+      const name_of_user = jwt.verify(token, process.env.jwt_secret)
```
Comment (joseluiz) [inline 05-JWT-Basics/preferred/middleware/authenticate.js]: Change the jwt_secret to uppercase. Example: process.env.JWT_SECRET

File: 05-JWT-Basics/preferred/controllers/main.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const jwt = require('jsonwebtoken')
+
+const logon = async (req, res) => {
+    const {username, password} = req.body
+
+    if(!username || !password) {
+        res.status(401).json({message: 'Provide Email and Password'})
+    }
+
+    const token = jwt.sign({username}, process.env.jwt_secret, {expiresIn: '90d'})
```
Comment (joseluiz) [inline 05-JWT-Basics/preferred/controllers/main.js]: Change the jwt_secret to uppercase. Example: process.env.JWT_SECRET

File: 05-JWT-Basics/preferred/public/browser-app.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+formDOM.addEventListener('submit', async (e) => {
+  formAlertDOM.classList.remove('text-success')
+  tokenDOM.classList.remove('text-success')
+
+  e.preventDefault()
+  const username = usernameInputDOM.value
+  const password = passwordInputDOM.value
+
+  try {
+    const { data } = await axios.post('/api/v1/login', { username, password })
```
Comment (joseluiz) [inline 05-JWT-Basics/preferred/public/browser-app.js]: This route doesn't exist in the preferred directory.

File: 05-JWT-Basics/preferred/public/browser-app.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  }
+  setTimeout(() => {
+    formAlertDOM.style.display = 'none'
+  }, 2000)
+})
+
+btnDOM.addEventListener('click', async () => {
+  const token = localStorage.getItem('token')
+  try {
+    const { data } = await axios.get('/api/v1/dashboard', {
```
Comment (joseluiz) [inline 05-JWT-Basics/preferred/public/browser-app.js]: This route doesn't exist in the preferred directory.

**Review by joseluiz — CHANGES_REQUESTED** [review]
Hey Rodgers. I left a few comments on things you need to fix. If you have any questions, feel free to call me on Slack.

File: 05-JWT-Basics/preferred/middleware/authenticate.js (hunk 1 above)
Comment (morarodgers) [reply 05-JWT-Basics/preferred/middleware/authenticate.js]: Done

File: 05-JWT-Basics/preferred/public/browser-app.js (hunk 3 above)
Comment (morarodgers) [reply 05-JWT-Basics/preferred/public/browser-app.js]: Fixed it

File: 05-JWT-Basics/preferred/public/browser-app.js (hunk 4 above)
Comment (morarodgers) [reply 05-JWT-Basics/preferred/public/browser-app.js]: Fixed it

**Comment by joseluiz** [comment]
Approved.

**Review by joseluiz — APPROVED** [review]
Approved

**Pull request merged**